### ISSUE TITLE
customer churn code

### ISSUE THREAD

**Issue by Debanjan65** [Debanjan65/college-project#3]
import gradio as gr
import pandas as pd
import joblib
import numpy as np

# Load saved model and scaler

model = joblib.load('airtel\_churn\_model.pkl')
scaler = joblib.load('scaler.pkl')

# Load data just to get feature names

df = pd.read\_csv(r"C:\Users\KIIT\Downloads\telecom\_churn.csv")
df.dropna(inplace=True)

# Encode object columns to match training process

from sklearn.preprocessing import LabelEncoder
label\_enc = LabelEncoder()
for col in df.select\_dtypes(include='object').columns:
if col != 'churn':
df\[col] = label\_enc.fit\_transform(df\[col])
if df\['churn'].dtype == 'object':
df\['churn'] = label\_enc.fit\_transform(df\['churn'])

# Get feature columns

X = df.drop('churn', axis=1)
feature\_names = X.columns.tolist()

# Create Gradio inputs dynamically

input\_components = \[]
for col in feature\_names:
if df\[col].dtype == 'int64' or df\[col].dtype == 'float64':
input\_components.append(gr.Number(label=col))
else:
input\_components.append(gr.Textbox(label=col))

# Prediction function

def predict\_churn(\*inputs):
input\_array = np.array(inputs).reshape(1, -1)
scaled\_input = scaler.transform(input\_array)
prediction = model.predict(scaled\_input)\[0]
return "Churned" if prediction == 1 else "Retained"

# Create Gradio Interface

gr.Interface(
fn=predict\_churn,
inputs=input\_components,
outputs=gr.Textbox(label="Prediction"),
title="Airtel Customer Churn Predictor",
description="Enter customer information to predict whether they will churn or not."
).launch()
make a presentation for this code in about 8-12 slides
